legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-generation/legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/testDataGeneration/testDataGeneration.pure

```diff
@@ -228,11 +228,25 @@ function meta::relational::testDataGeneration::generateSeedDataString(func:Funct
    '[\n'+$result.values->at(0).rows->map(r|'       meta::relational::testDataGeneration::createRowIdentifier(['+$mainTable.primaryKey->map(c|$c.name)->map(c|'\''+$c+'\'')->joinStrings(',')+'], ['+$r.values->map(c|$c->toRepresentation())->makeString(',')+'])')->sort()->joinStrings(',\n')+'\n  ])\n';
 }
 
-function meta::relational::testDataGeneration::generateParametersNameValuePair(func: FunctionDefinition<Any>[1], parameters: Any[*]):Pair<String, List<Any>>[*]
+function meta::relational::testDataGeneration::generateParametersNameValuePair(func: FunctionDefinition<Any>[1], parameters: Any[*], parameterNameValuePairs: Pair<String, List<Any>>[*]):Pair<String, List<Any>>[*]
 {
    let params = $func->functionType().parameters->evaluateAndDeactivate();
    assertEquals($params->size(), $parameters->size(), | 'Missing some values/Extra values provided for parameters ['+$params.name->joinStrings(', ')+']');
-   let newVars = $params->size()->range()->map(i | pair($params->at($i).name, list(if($params->at($i).genericType.rawType->toOne()->_subTypeOf(Date) && $parameters->at($i)->instanceOf(String), | $parameters->at($i)->cast(@String)->parseDate(), | $parameters->at($i)))));
+   let newVars = $params->size()->range()->map(i | let paramType = $params->at($i).genericType.rawType->toOne();
+                                                   let isDate = if($parameterNameValuePairs->size() > 0 ,
```

Comment:
think we should change `->size() > 0 ` to `isNotEmpty()`

Reply:
updated! thanks!